### PR TITLE
fix(gar): prune gateways immediately once they hit 30 consecutive fai…

### PR DESCRIPTION
**Pull request by dtfiedler**
…lures

### CODE REVIEW

File: src/main.lua [hunk 1]
```diff
@@ -1754,6 +1762,10 @@ addEventingHandler("distribute", utils.hasMatchingTag("Action", "Tick"), functio
 	if #newDemandFactors > 0 then
 		msg.ioEvent:addField("New-Demand-Factors", newDemandFactors, ";")
 	end
+	if #newPrunedGateways > 0 then
```
[the remark this answers is not in the document]
Comment (dtfiedler) [reply src/main.lua:1765]: we'll actually need to redo all of that here - as it has the same potential impacts for supply 😢

File: src/main.lua [hunk 2]
```diff
@@ -1655,18 +1655,22 @@ addEventingHandler("distribute", utils.hasMatchingTag("Action", "Tick"), functio
...
 		local distributedEpoch = epochs.distributeRewardsForEpoch(timestamp)
 		if distributedEpoch ~= nil and distributedEpoch.epochIndex ~= nil then
 			tickedRewardDistributions[tostring(distributedEpoch.epochIndex)] =
 				distributedEpoch.distributions.totalDistributedRewards
 			totalTickedRewardsDistributed = totalTickedRewardsDistributed
 				+ distributedEpoch.distributions.totalDistributedRewards
 		end
-
+		-- prune any gateway that has hit the failed 30 consecutive epoch threshold after the epoch has been distributed
+		local prunedGateways = gar.pruneGateways(timestamp)
```
Comment (arielmelendez) [inline src/main.lua:1667]: This returns more than just gateways:
image

Comment (dtfiedler) [reply src/main.lua:1667]: ah - right, here we would actually only care about `slashedGateways` and `stakeSlashed`. no delegate stake would be returned (as it was already computed and done by the `prune` that is called before this).

conversely, in `prune` we should not need to account for `stakeSlashed` or `slashedGateways` as that should will happen on this handler, not in the `prune`

**Review by arielmelendez — CHANGES_REQUESTED** [review]
I'm going to push a commit to properly track the circulating supply changes here.

**Review by arielmelendez — APPROVED** [review]

**Pull request merged**